strawberry/extensions/context.py

```diff
@@ -0,0 +1,67 @@
+from abc import ABC
+from typing import List
+
+from strawberry.extensions import Extension
+from strawberry.utils.await_maybe import await_maybe
+
+
+class ExtensionContextManager(ABC):
+    enter_method: str
+    exit_method: str
+
+    def __init__(self, extensions: List[Extension]):
+        self.extensions = extensions
+
+
+class RequestContextManager(ExtensionContextManager):
+    def __enter__(self):
+        for extension in self.extensions:
+            extension.on_request_start()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for extension in self.extensions:
+            extension.on_request_end()
+
+    async def __aenter__(self):
+        for extension in self.extensions:
+            await await_maybe(extension.on_request_start)
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        for extension in self.extensions:
+            await await_maybe(extension.on_request_end)
```

Comment:
Hmm, this feels even more verbose than before... Is there a reason that the `Extension` itself isn't its own `ContextManager`?

Reply:
Would that simplify the code? This is also something for another PR, I think I tried to find a nice way for handling extensions, but I failed :D